File: Maps_Project.Web/Modules/Trip/BusinessTripDetail/BusinessTripDetailRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;
using System.IO;

namespace Maps_Project.Trip;

[ConnectionKey("Default"), Module("Trip"), TableName("BusinessTripDetail")]
[DisplayName("Business Trip Detail"), InstanceName("Business Trip Detail"), GenerateFields]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
public sealed partial class BusinessTripDetailRow : Row<BusinessTripDetailRow.RowFields>, IIdRow
{
    const string jBusinessTrip = nameof(jBusinessTrip);

    [DisplayName("Id"), Identity, IdProperty]
    public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Business Trip"), ForeignKey(typeof(BusinessTripRow)), LeftJoin(jBusinessTrip), TextualField(nameof(BusinessTripTitle))]
    [ServiceLookupEditor(typeof(BusinessTripRow))]
    public int? BusinessTripId { get => fields.BusinessTripId[this]; set => fields.BusinessTripId[this] = value; }

    [DisplayName("Latitude")]
    public decimal? Latitude { get => fields.Latitude[this]; set => fields.Latitude[this] = value; }


    [DisplayName("Longitude")]
    public decimal? Longitude { get => fields.Longitude[this]; set => fields.Longitude[this] = value; }

    [DisplayName("Business Trip Title"), Origin(jBusinessTrip, nameof(BusinessTripRow.Title))]
    public string BusinessTripTitle { get => fields.BusinessTripTitle[this]; set => fields.BusinessTripTitle[this] = value; }
}

File: Maps_Project.Web/Modules/ServerTypes/Trip/BusinessTripDetailRow.ts
import { fieldsProxy } from "@serenity-is/corelib";

export interface BusinessTripDetailRow {
    Id?: number;
    BusinessTripId?: number;
    Latitude?: number;
    Longitude?: number;
    BusinessTripTitle?: string;
}

export abstract class BusinessTripDetailRow {
    static readonly idProperty = 'Id';
    static readonly localTextPrefix = 'Trip.BusinessTripDetail';
    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<BusinessTripDetailRow>();
}